inst/tests/tests.Rraw

```diff
@@ -6968,8 +6968,8 @@ test(1497, DT[, .SD, .SDcols = !c("a", "c")], DT[, !c("a", "c"), with=FALSE])
 
 # Fix for #1060
 DT = data.table(x=1, y=2, z=3, a=4, b=5, c=6)
-test(1498.1, DT[, .SD, .SDcols=c(TRUE,FALSE)], DT[, c("x", "z", "b"), with=FALSE])
-test(1498.2, DT[, .SD, .SDcols=!c(TRUE,FALSE)], DT[, !c("x", "z", "b"), with=FALSE])
+test(1498.1, DT[, .SD, .SDcols=c(TRUE,FALSE)], DT[, c("x", "z", "b"), with=FALSE], warning = "When .SDcols is a logical vector, each column should have a TRUE or FALSE entry. Vector will be repeated to length of data.table. Warning will change to error in year 2022.")
```

Comment:
we are usually doing a deprecation cycle rather than a fixed date, e.g. it becomes warning on this release, then error on next release

Reply:
I like dates because it's more tangible to the end-user. But I assume this would be warning until 1.14 and then make it as an error?